### ISSUE TITLE
Usage examples

### ISSUE THREAD

**Issue by ilanl** [doowb/async-helpers#21]
Hi,

I'm missing some documentation on how to register async helper. I'm using Typescript.

data: { ids: ["1", "2", "3" ]}

template: 
`{{{ listConcat (translate ./ids) 'AND' ' ' }}}`

expected: to return a readable list of translated ids like so
Banana, Orange and Pears

listConcat is a helper that creates a readable list separated by ", " and "AND/OR"
translate is an async helper that returns the translated string|string[] for a given id or list of ids, it takes into account an extra parameter to understand the target language (this is not part of the template but part of the bigger context) 

```
const extraContext = { config: { lang: input.lang } }
const rendered = template(data, extraContext);
```

Helpers:
```
Handlebars.registerHelper({
    listConcat: function (items: string[], special: string, joinWith: string) {
        const mapItems = ([] as string[]).concat(...items.map((item, i, arr) => {
          if (i === 0) {
            return [item]
          }
          else if (i === arr.length-1) {
            return [' ', special, joinWith, item]
          }
          return [', ', item] 
        }));
        return mapItems.join('');
    },
    translate: function(args: string[] | string, extraContext) {
        const { config } = extraContext; // used to pass the current language to translate to
        const fn = (id: string) => getTranslationForItem({itemId: id, itemType: 'product'}, config.lang)
        if (Array.isArray(args)) {
            let result = args.map((i: string) => fn(i));
            return result;
        }
        else {
            return fn(args);
        }
    }
});
```
getTranslationForItem is async and I'm trying to make it work.

Any help is appreciated! I like the simplicity of handlebars but it's unclear how to support async helpers

**Comment by doowb** [comment]
Hi @ilanl ...

## TL;DR;

Handlebars doesn't support async helpers natively, that's why we created this package. See [the example](https://github.com/doowb/async-helpers/blob/master/examples/example.js) for how it could be used.

## More details

The example uses `async-helpers` to wrap Handlebars helpers with async functionality, then registers the wrapped helpers with `Handlebars.registerHelper`. At the end of the example, `resolve` is used to "resolve" async ids in the first pass of the rendered output.

There are basically 2 "rendering" passes happening because Handlebars has to be sync and our wrapped helper functions create unique ids that get rendered into the output. The second pass is async and handles finding and resolving the ids by executing the helpers as async functions. The ids are replaced with the results from the async helpers.

I have a longer explanation in an issue/pr some place that I'll try to find and link to later.

BTW... the example is complicated and has specific code in there for handling Handlebars partials that may be async... most of this logic is implemented in some of our other packages to make `async-helpers` simpler to work with. Also, as Jon mentioned in another PR comment, the `async-helpers` logic has also been refactored and is easier to use and handles Promises, but that hasn't been implemented in this package yet.
